payara-micro/src/main/java/fish/payara/arquillian/container/payara/managed/PayaraMicroContainerConfiguration.java

```diff
@@ -199,15 +199,22 @@ public void validate() throws ConfigurationException {
             throw new IllegalArgumentException("Could not locate the Payara Micro Jar file " + getMicroJar());
         }
 
+        String zipEntryFileDir = ""; //Created here for the sakes of debugging in case of NPE
+        
         try (JarFile microJarFile = new JarFile(getMicroJarFile())) {
-            ZipEntry pomProperties = microJarFile
-                    .getEntry("META-INF/maven/fish.payara.micro/payara-micro-boot/pom.properties");
+            
+            zipEntryFileDir = "MICRO-INF/domain/branding/glassfish-version.properties";
+            ZipEntry pomProperties = microJarFile.getEntry(zipEntryFileDir);
+
             Properties microProperties = new Properties();
             microProperties.load(microJarFile.getInputStream(pomProperties));
-            this.microVersion = new PayaraVersion(microProperties.getProperty("version"));
+            this.microVersion = new PayaraVersion(microProperties.getProperty("major_version") + "." + microProperties.getProperty("minor_version"));
```

Comment:
Missing `update_version`, `payara_version` and `payara_update_version` as in the case of Payara 4.1.2.191.4 following properties are listed too in `glassfish-version.properties`.

    major_version=4
    minor_version=1
    update_version=2
    payara_version=191
    payara_update_version=4

Reply:
Do you mean payara 4 micro? 😉 I don't think we'll return to this versioning scheme, so major and minor are enough

Reply:
Payara 4 is an example in the above comment, https://github.com/payara/Payara/blob/master/appserver/packager/appserver-base/src/main/resources/config/branding/glassfish-version.properties includes all of the above properties

Another example:

    major_version=5
    minor_version=191
    update_version=5

Reply:
So it looks like PayaraVersion needs a factory method `fromBrandingProperties` then, because this is getting complicated.

Reply:
I think I want to mark this as a void point, I'm open to discussion on it but as far as I'm aware Payara 4 versions aren't affected by the package changes in core server and so wouldn't (arguably shouldn't) use the newest container release; they could stick on 1.1/1.2

Reply:
Made an implementation to accommodate this - all input is very welcome